src/java.base/share/classes/java/lang/StringConcatHelper.java

```diff
@@ -368,16 +368,12 @@ static String simpleConcat(Object first, Object second) {
             // newly created string required, see JLS 15.18.1
             return new String(s1);
         }
-        // start "mixing" in length and coder or arguments, order is not
-        // important
-        long indexCoder = mix(initialCoder(), s1);
-        indexCoder = mix(indexCoder, s2);
-        byte[] buf = newArray(indexCoder);
-        // prepend each argument in reverse order, since we prepending
-        // from the end of the byte array
-        indexCoder = prepend(indexCoder, buf, s2);
-        indexCoder = prepend(indexCoder, buf, s1);
-        return newString(buf, indexCoder);
+        byte coder = (byte) (s1.coder() | s2.coder());
```

Comment:
```suggestion
        int coder = s1.coder() | s2.coder();
```

Reply:
Converting coder to int type requires processing when calling new String(byte[], byte)

Reply:
You are right, I ignored that `coder` is reused again below. My bad!